_documentation/03-designing-scanner-drivers.md

```diff
@@ -2,5 +2,103 @@
+
+## Introduction
+
+A driver is a code or data specific to a certain model or group of hardware devices, needed to make the hardware work with the hardware-model-independent code of the operating system. Printing in Linux has moved towards [Driverless Printing](https://openprinting.github.io/driverless/) and so does Scanning, which means there is no need for any hardware-model-specific code or data. 
+
+Kindly refer to <a href="../02-designing-printer-drivers/">Tutorial to Design Printer Drivers</a> know more about Printer Applications and their designing. Designing Scanner Applications is very much similar and expects the manufacturers to use PAPPL framework to reduce their effort in implementing all things from scratch.
+
+## Working of PAPPL-based Scanner Driver
+
+PAPPL provides support for the below mentioned commands by default:
+
+* add
+* cancel
+* default
+* delete
+* devices
+* drivers
+* jobs
+* modify
+* options
+* printers
+* scan
+* server
+* shutdown
+* status
+* submit
+
+Once the User issues any of the following Sub-command, The printer/scanner application converts these commands (or corresponding IPP requests) into the necessary USB/network/etc. communications to perform a scan, get capabilities, and so forth.
+
+The class of device that we are supporting for PAPPL, the scanner will only produce raster data. It might provide native `TIFF` support, but that is probably about it.
+
+The PAPPL scan interface should pull image data for the current page in one of three formats: 1-bit grayscale, 8-bit grayscale, and 24-bit sRGB. It will then be up to PAPPL to deliver the pages to the Client in `PNG`, `JPEG`, `TIFF`, or `PDF` format.  For `PNG` and `JPEG`, each page is a separate document in the scan job.  For `TIFF` and `PDF`, a single document is generated and streamed to the Client in response to the Get-Next-Document-Data operation.
+
+<p align="center">
+    <img src="../../assets/images/scanner.png"/>
+</p>
+
+The scanner interface returns a single page at a time as uncompressed (or losslessly compressed) data, and PAPPL will convert that to a handful of formats requested by the Client.  In the case of IPP Scan, `PDF` and `JPEG` are required and PAPPL should support `PNG` for lossless compression of single images and `TIFF` for multi-page images.
+
+## Designing Components
+
+PAPPL provides callbacks for most of the events including submitting print/scan jobs, querying printer/scanner status and capabilities, and so forth to reduce the manufacturer's job and hence designing scanner applications is very much similar to designing printer application. The only difference if the additional callback support provided for the scan command. 
```

Comment:
PAPPL provides callbacks for most of the events including submitting print/scan jobs, querying printer/scanner status and capabilities, and so forth to reduce the manufacturer's **workload** and hence designing scanner applications is very much similar to designing printer application**s**. The only difference **is** the additional callback support provided for the scan command.